src/Dapr.Actors/Runtime/ActorTestOptions.cs

```diff
@@ -0,0 +1,97 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Dapr.Actors.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Dapr.Actors.Runtime
+{
+    /// <summary>
+    /// Specifies optional settings settings for the <see cref="ActorHost" /> when creating an actor
+    /// instance for testing.
+    /// </summary>
+    public sealed class ActorTestOptions
+    {
+        /// <summary>
+        /// Gets or sets the <see cref="ActorId" />.
+        /// </summary>
+        /// <returns></returns>
+        public ActorId ActorId { get; set; } = ActorId.CreateRandom();
+
+        /// <summary>
+        /// Gets or sets the <see cref="ILoggerFactory" />.
+        /// </summary>
+        /// <value></value>
+        public ILoggerFactory LoggerFactory { get; set; } = NullLoggerFactory.Instance;
+
+        /// <summary>
+        /// Gets or sets the <see cref="JsonSerializerOptions" />.
+        /// </summary>
+        public JsonSerializerOptions JsonSerializerOptions { get; set; } = new JsonSerializerOptions(JsonSerializerDefaults.Web);
+
+        /// <summary>
+        /// Gets or sets the <see cref="IActorProxyFactory" />.
+        /// </summary>
+        public IActorProxyFactory ProxyFactory { get; set; } = new InvalidProxyFactory();
+
+        /// <summary>
+        /// Gets or sets the <see cref="ActorTimerManager" />.
+        /// </summary>
+        public ActorTimerManager TimerManager { get; set; } = new InvalidTimerManager();
+
+        private class InvalidProxyFactory : IActorProxyFactory
```

Comment:
Note, it's now also possible to mock the proxy factory. So you can mock calls to other actors.